Docs edits (#592)
opentensor/developer-docs: docs/miners/index.md

To participate as a miner, you must first register a hotkey with the subnet in order to receive a UID on that subnet.

:::tip No need to create a subnet to mine
You **do not** have to create a subnet to mine on the Bittensor network. Most miners work on already established subnets.
:::

Registration has a cost in TAO, which fluctuates dynamically based on the time since the last registration. When you secure a UID slot in a subnet on the main chain, this TAO is sunk cost and cannot be recovered.

A subnet can have a maximum of 64 subnet validator UIDs and 192 subnet miner UIDs (256 total) in subnets other than Subnet 1.

Upon registration, your hotkey, which is part of your wallet, becomes the holder of the UID slot.

:::tip Ownership belongs to a hotkey

> When you delegate your TAO to a subnet validator, you attach your delegated TAO to that validator’s hotkey. See [Delegation](../staking-and-delegation/delegation.md).

Reviewer: whoops, looks like this line got doubled
```suggestion
```

When you delegate your TAO to a subnet validator, you attach your delegated TAO to that validator’s hotkey. See [Delegation](../staking-and-delegation/delegation.md).

A hotkey can hold multiple UIDs across **separate** subnets. However, within one subnet, each UID must have a unique hotkey.
:::

To register your keys with a subnet, run the following command on your terminal, replacing `<your_preferred_netuid>`, `<my_coldkey>`, `<my_hotkey>`.
`<your_preferred_netuid>` is the `netuid` of your preferred subnet.

```bash
btcli subnet register --netuid <your_preferred_netuid>  --wallet.name  <my_coldkey> --wallet.hotkey <my_hotkey>
```

For example, to register your keys with subnet 1—netuid of 1:

```bash
btcli subnet register --netuid 1 --wallet.name test-coldkey --wallet.hotkey test-hotkey
```

## Miner deregistration

> A subnet miner can be deregistered if its performance is poor. Mining is competitive—and the UID slots are limited. Aside from Subnet 1, all subnets have 256 UID slots per subnet. Of these 256 UID slots, a subnet can have a maximum of 64 subnet validator UIDs and 192 subnet miner UIDs. Each tempo, the lowest ranked miners risk being replaced by a newly registered miner, who takes over that UID.

Reviewer: ```suggestion
A miner can be deregistered if it earns low emissions due to receiving low weights (ratings) from validators. Typical subnets have 256 UID slots per subnet, of which a maximum of 64 subnet can be occupied by validators. Each tempo, the lowest ranked slot is deregistered from the hotkey that holds it and assigned to a new registrant.
```

- Every subnet has an `immunity_period` hyperparameter expressed in a number of blocks.
  :::tip See
  See [`immunity_period`](../subnets/subnet-hyperparameters.md#immunityperiod).
  :::
- A subnet miner or validator at a UID (in that subnet) has a defined number of blocks to improve its performance. This is known as `immunity_period`. When the `immunity_period` expires, that miner or validator can be deregistered if it has the lowest performance in the subnet and a new registration arrives.
- A neuron's `immunity_period` starts when the miner or validator is registered into the subnet.

>   Miners as well as validators can be deregistered if their emissions are low.

Reviewer: ```suggestion
Validators as well as miners can be deregistered if their emissions are low; either role requires a UID.
```
